build-tools/packages/build-cli/test/commands/generate/buildVersion.test.ts

```diff
@@ -345,3 +345,292 @@ describe("generate:buildVersion", () => {
 			expect(ctx.stdout).to.contain("isLatest=false");
 		});
 });
+
+describe("generate:buildVersion for alpha/beta", () => {
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "prerelease",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "alpha",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("client tag with package types field selected as alpha", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.4.0-88879-alpha-types");
+			expect(ctx.stdout).to.contain("isLatest=false");
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "prerelease",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "beta",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("client tag with package types field selected as beta", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.4.0-88879-beta-types");
+			expect(ctx.stdout).to.contain("isLatest=false");
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "prerelease",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "none",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("client tag with package types field selected as none", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.4.0-88879");
+			expect(ctx.stdout).to.contain("isLatest=false");
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "prerelease",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "public",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("client tag with package types field selected as public", (ctx) => {
```

Comment:
It might be easier to compare the test results if you use a more structured format for the names, e.g.

```suggestion
		.it("tagname: client, release: prerelease, types: public", (ctx) => {
```